v1/src/main/java/com/google/cloud/teleport/spanner/ddl/InformationSchemaScanner.java

```diff
@@ -272,13 +273,26 @@ private void listTables(Ddl.Builder builder) {
       // Parent table and child table has to be in same schema.
       String parentTableName =
           resultSet.isNull(2) ? null : getQualifiedName(tableSchema, resultSet.getString(2));
-      String onDeleteAction = resultSet.isNull(3) ? null : resultSet.getString(3);
+      String interleaveTypeStr = resultSet.isNull(3) ? null : resultSet.getString(3);
+      Table.InterleaveType interleaveType = null;
+      if (!Strings.isNullOrEmpty(interleaveTypeStr)) {
+        interleaveType =
+            interleaveTypeStr.equals("IN PARENT") ? InterleaveType.IN_PARENT : InterleaveType.IN;
+      }
+      String onDeleteAction = resultSet.isNull(4) ? null : resultSet.getString(4);
 
       // Error out when the parent table or on delete action are set incorrectly.
-      if (Strings.isNullOrEmpty(parentTableName) != Strings.isNullOrEmpty(onDeleteAction)) {
+      if (Strings.isNullOrEmpty(parentTableName) != Strings.isNullOrEmpty(interleaveTypeStr)) {
```

Comment:
Can you explain why this is not a valid combination?

parentTableName is not null && interleaveTypeStr is null  -> This would be the current situation in prod right? Or is the interleaveType column in info schema is fully backfilled with `IN PARENT` ?

Reply:
See my above comment, but no, if parentTableName is not null interleaveTypeStr will never be null, even before I released my feature.

As I understand it, the only place backwards compatibility is relevant is the avro -> ddl converter, in case we try to import from an export performed before this change. There, it is important that we default to NO ACTION in case the property doesnt exist.